### PR TITLE
Publish to docker hub

### PR DESCRIPTION
**Pull request by chriswininger**
this pr is no where near ready but I need to make it a pr to test as per https://stackoverflow.com/questions/63362126/github-actions-how-to-run-a-workflow-created-on-a-non-master-branch-from-the-wo

### CODE REVIEW

File: api-server/docker-publish.sh [hunk 1]
```diff
@@ -0,0 +1,19 @@
+#!/usr/bin/env bash
+
+set -e
+
+tag="${1:=local}"
+container_name="infinite-api-server"
+dockerhub_org="chriswininger"
```
Comment (MatthewGidcomb) [inline api-server/docker-publish.sh:7]: Could read the org name from an env variable, and then put it in a Github secret; then if/when we create an Infinite Dockerhub org we just have to change it in one place.

File: .github/workflows/publish-to-docker-hub.yml [hunk 2]
```diff
@@ -0,0 +1,50 @@
+on:
+  workflow_dispatch:
+    environment: docker-publication
```
Comment (MatthewGidcomb) [inline .github/workflows/publish-to-docker-hub.yml:3]: What does using an environment here give us?

File: api-server/docker-publish.sh (hunk 1 above)
Comment (chriswininger) [reply api-server/docker-publish.sh:7]: I don't think it really needs to be a secret. Nothing secret about it and this keeps the script flexible for local host, at least that was my thinking

File: .github/workflows/publish-to-docker-hub.yml (hunk 2 above)
Comment (chriswininger) [reply .github/workflows/publish-to-docker-hub.yml:3]: well the TEST_SECRET one I need to delete but the COMMIT_HASH one just makes it availabe in the script for

`./docker-build.sh "$COMMIT_HASH"`

I think I could also just do:

`./docker-buid.sh {inputs.commit}`

I'm honestly not sure which is better but the tutorial I followed mapped the inputs and secrets to env for some reason, so really just copy/paste

**Review by MatthewGidcomb — APPROVED** [review]

**Pull request merged**